### ISSUE TITLE
[SPIR-V] Trying to "write" to uniforms using the -Gec backwards compatibility mode enabled produces invalid codegen (works with DXIL backend)

### ISSUE THREAD

**Issue by Dredhog** [microsoft/DirectXShaderCompiler#3890]
### Title
[SPIR-V] Trying to "write" to uniforms using the `-Gec` backwards compatibility mode enabled produces invalid codegen (works with DXIL backend)
 
### Functional impact
Old HLSL shaders written for FXC which rely on the `-Gec` option don't compile with DXC's SPIR-V backend. DXIL backend does not complain and the generated shader appears to render correctly.

### Minimal repro steps

1. Compile the following shader with `dxc.exe -T ps_6_0 -E CSMain -spirv repro_shader.txt`:
```
float4 myGlobal; 
 
RWStructuredBuffer<float4> outBuffer; 
[numthreads(8, 8, 1)] 
void CSMain(uint3 id : SV_DispatchThreadID) : SV_TARGET 
{ 
    if(myGlobal.x < .5) 
        myGlobal += float4(id.xyzx); 
    outBuffer[id.x] = myGlobal; 
} 
```
2. Observe the compiler output:

```
warning: /Gec flag is a deprecated functionality.
fatal error: generated SPIR-V is invalid: In the Vulkan environment, cannot store to Uniform Blocks
  OpStore %26 %35

note: please file a bug report on https://github.com/Microsoft/DirectXShaderCompiler/issues with source code if possible
```

### Expected result
Compilation completes without validation errors. The implementation should probably create a temporary local variable when a write to a uniform is performed (this is what FXC is doing, did not read the DXIL disassembly but it works at runtime)

### Actual result
Compilation fails with a fatal error, asking to create a file report. The generated SPIR-V is likely trying to write directly into the UBO.

### Further technical details
Output copied from [shader playground](http://shader-playground.timjones.io/350f246b48a4efa5ad578a0518558f06). Also get the same result with the [June 2021 release](https://github.com/microsoft/DirectXShaderCompiler/releases#:~:text=Compare-,DX%20Compiler%20release%20for%20June%202021,-hekota%20released%20this)

**Comment by jaebaek** [comment]
Thank you for reporting this issue. We will take a look!

**Comment by sudonatalie** [comment]
Given that this option is already deprecated in the DXIL backend, we don't have any plans to add support for it in the SPIR-V backend, but we should make that error messaging explicit.
